src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -39,14 +38,18 @@ gl_sframe get_data(const data_iterator::parameters& params) {
     images = images.apply(image_util::encode_image, flex_type_enum::IMAGE);
   }
 
-  gl_sframe result({ { params.annotations_column_name, annotations },
-                     { params.image_column_name,       images      }  });
+  gl_sframe result({{params.image_column_name, images}});
 
   if (!params.predictions_column_name.empty()) {
     result[params.predictions_column_name] =
         params.data[params.predictions_column_name];
   }
 
+  if (!params.annotations_column_name.empty()) {
+    result[params.annotations_column_name] =
+        params.data[params.annotations_column_name];
+  }
+
```

Comment:
Nit: let's handle the annotations case before the predictions case, to parallel the order in the parameters (and the order of importance)

Reply:
okay